hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoApproveAllowanceTransitionLogic.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * -
+ *  * ‌
+ *  * Hedera Services Node
+ *  * ​
+ *  * Copyright (C) 2018 - 2022 Hedera Hashgraph, LLC
+ *  * ​
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *      http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *  * ‍
+ *
+ */
+
+package com.hedera.services.txns.crypto;
+
+import com.hedera.services.context.TransactionContext;
+import com.hedera.services.ledger.SigImpactHistorian;
+import com.hedera.services.state.submerkle.FcTokenAllowance;
+import com.hedera.services.state.submerkle.FcTokenAllowanceId;
+import com.hedera.services.store.AccountStore;
+import com.hedera.services.store.models.Account;
+import com.hedera.services.store.models.Id;
+import com.hedera.services.txns.TransitionLogic;
+import com.hedera.services.txns.crypto.validators.AllowanceChecks;
+import com.hedera.services.utils.EntityNum;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.CryptoAllowance;
+import com.hederahashgraph.api.proto.java.NftAllowance;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.TokenAllowance;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import static com.hedera.services.exceptions.ValidationUtils.validateFalse;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.MAX_ALLOWANCES_EXCEEDED;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.SUCCESS;
+
+public class CryptoApproveAllowanceTransitionLogic implements TransitionLogic {
+	private final TransactionContext txnCtx;
+	private final SigImpactHistorian sigImpactHistorian;
+	private final AccountStore accountStore;
+	private final AllowanceChecks allowanceChecks;
+
+	// Should these be dynamic properties?
```

Comment:
Anytime I've answered "no" to that question, I've regretted it eventually. 😀

Reply:
haha got it ! 😄 Made them dynamic properties using these names `hedera.allowances.maxTransactionLimit` , `hedera.allowances.maxAccountLimit`. Is there any naming convention that should be followed @tinker-michaelj  ?